hooks/pre-exit

```diff
@@ -2,17 +2,13 @@
 
 set -eo pipefail
 
-if [[ "$BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_SKIP_CHECKOUT" == "true" ]]; then
-    exit 0
+if [[ "$BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_SKIP_CHECKOUT" == "true" \
+    || "$BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_DELETE_CHECKOUT" != "true" ]]; then
```

Comment:
What's the rationale behind not deleting checked out code by default?

Reply:
That's the behavior of the default checkout, so it seemed like the sensible default